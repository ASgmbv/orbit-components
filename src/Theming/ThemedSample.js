// @flow
import React from "react";
import { withTheme } from "theming";

import defaultTheme from "./defaultTheme";

// const defaultThemeProps = {
//   color: defaultTheme.color,
//   fontFamily: defaultTheme.fontFamily,
// };

type ComponentProps = {
  // theme: $Shape<typeof defaultTheme>,
  theme: typeof defaultTheme,
};

const Component = ({ theme }: ComponentProps) => (
  <div style={{ color: theme.color, fontFamily: theme.fontFamily }}>{theme.color}</div>
);

// We're passing a default theme for Component that aren't wrapped in the ThemeProvider
Component.defaultProps = {
  theme: defaultTheme, // eslint-disable-line react/default-props-match-prop-types
};

const ThemedComponent = withTheme(Component);
export { ThemedComponent as default, Component as RawComponent };
