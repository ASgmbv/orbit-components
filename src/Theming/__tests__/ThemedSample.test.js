// @flow

import * as React from "react";
import { shallow } from "enzyme";

import { RawComponent } from "../ThemedSample";
// import defaultTheme from "../defaultTheme";

describe("RawComponent", () => {
  const component = shallow(<RawComponent />);
  const div = component.find("div");

  it("should match snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  it("Should contain a correct style", () => {
    expect(div.render().attr("style")).toBe("color:blue;font-family:Courier");
    // expect(div.render().attr("style").fontFamily).toBe(defaultTheme.fontFamily);
  });

  it("Should contain a color", () => {
    expect(div.render().text()).toBe("blue");
  });
});
