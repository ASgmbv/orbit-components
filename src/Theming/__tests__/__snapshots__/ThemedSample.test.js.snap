// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RawComponent should match snapshot 1`] = `
<div
  style={
    Object {
      "color": "blue",
      "fontFamily": "Courier",
    }
  }
>
  blue
</div>
`;
